pkg/asset/machines/azure/machines.go

```diff
@@ -104,14 +110,29 @@ func provider(platform *azure.Platform, mpool *azure.MachinePool, osImage string
 			},
 		},
 		Zone:            az,
-		Subnet:          fmt.Sprintf("%s-%s-subnet", clusterID, role),
+		Subnet:          subnet,
 		ManagedIdentity: fmt.Sprintf("%s-identity", clusterID),
-		Vnet:            fmt.Sprintf("%s-vnet", clusterID),
-		ResourceGroup:   fmt.Sprintf("%s-rg", clusterID),
+		Vnet:            virtualNetwork,
+		ResourceGroup:   networkResourceGroup,
```

Comment:
there are 2 fields, ResourceGroup where the machine needs to be created and NetworkResourceGroup where the Vnet, subnet are...

https://github.com/openshift/cluster-api-provider-azure/blob/719f40a46924bb1a46143a8df291b33ec17075b8/pkg/apis/azureprovider/v1beta1/azuremachineproviderconfig_types.go#L81-L82

Reply:
`ResourceGroup` stays `fmt.Sprintf("%s-rg", clusterID),`